spec/Section 6 -- Execution.md

```diff
@@ -143,6 +146,120 @@ ExecuteMutation(mutation, schema, variableValues, initialValue):
+observed. An observer of an event stream may cancel observation to avoid stop
+future events.
+
+If the operation is a subscription, the result is an event stream called the
+"Publish Stream" where each event in the stream is called a "Publish Payload".
+
+#### Subscribe
+
+Executing a subscription creates a persistent function on the server that
+maps an underlying event stream to the Publish Stream. The logic to create the
+underlying event stream is domain-specific and takes the root field and query
+variables as inputs.
+
+Subscribe(schema, subscription, operationName, variableValues, initialValue):
+
+  * Let {subscriptionType} be the root Subscription type in {schema}.
+  * Assert: {subscriptionType} is an Object type.
+  * Let {selectionSet} be the top level Selection Set in {subscription}.
+  * Let {rootField} be the first top level field in {selectionSet}.
+  * Let {eventStream} be the result of running {CreateUnderlyingEventStream(rootField, variableValues)}.
+  * Let {publishStream} be the result of running {MapEventToPayload(eventStream)}
+
+CreateUnderlyingEventStream(rootField, variableValues):
+
+  * *Application-specific logic to map from root field/variables to events*
+
+#### Publish Stream
+
+Each event in the underlying event stream triggers execution of the subscription
+selection set.
+
+MapEventToPayload(eventStream):
+
+  * For each {event} on {eventStream}:
+    * Let {publishPayload} be the result of running
+      {ExecuteSelectionSet(selectionSet, subscriptionType, event, variableValues)}
+      *normally* (allowing parallelization).
+    * Let {errors} be any *field errors* produced while executing the
+      selection set.
+    * Yield an unordered map containing {publishPayload} and, optionally,
+      {errors} on {publishStream}.
+  * At any time while the publish stream is active, the client or server may
+  Unsubscribe().
+
+Common reasons for Unsubscribe() include:
+  * client no longer wants to receive payloads for a subscription.
+  * The underlying event stream has produced an error or has naturally ended.
+
+#### Unsubscribe
+
+Unsubscribe cancels the Publish Stream. This is also a good opportunity for the
+server to clean up the underlying event stream and any other resources used by
+the subscription.
+
+Unsubscribe()
+
+  * Cancel {publishStream}
+
+#### Example
+
+As an example, consider a chat application. To subscribe to new messages posted
+to the chat room, the client sends a request like so:
+
+```graphql
+subscription NewMessages {
+  newMessage(roomId: 123) {
+    sender
+    text
+  }
+}
+```
+
+While the client is subscribe, whenever new messages are posted to chat room
+with ID "123", the selection for "sender" and "text" will be evaluated and
+published to the client, for example:
+
+```js
+{
+  "data": {
+    "newMessage": {
+      "sender": "Hagrid",
+      "text": "You're a wizard!"
+    }
+  }
+}
+```
+
+#### Recommendations and Considerations for Supporting Subscriptions
```

Comment:
This is a long section title and it's specifically about scale. So perhaps just `Supporting Subscriptions at Scale`?